backend/app/services/workflow_engines/reana.py

```diff
@@ -115,112 +112,113 @@ async def run_workflow(self, experiment_run: ExperimentRun) -> WorkflowState:
-                )
-
-                savepath_file = os.path.join(
-                    local_save_dirpath, *matched_file["name"].split("/")
-                )
-                os.makedirs(os.path.dirname(savepath_file), exist_ok=True)
-                with open(savepath_file, "wb") as f:
-                    f.write(binary)
-
         except ReanaConnectionException as e:
             raise e
         except Exception:
-            # TODO: Handle exception properly
-            pass
+            return None
 
-    async def save_metadata(
-        self, experiment_run: ExperimentRun, workflow_state: WorkflowState
-    ) -> bool:
-        workflow_name = self.get_workflow_name(experiment_run)
-        experiment_dirpath = settings.get_experiment_run_path(run_id=experiment_run.id)
+        savedir.mkdir(parents=True, exist_ok=True)
+        filename = os.path.basename(f"{filepath}.zip" if zipped else filepath)
 
-        try:
-            # retrieve logs
-            logs = (
-                await self._call_reana_function(
-                    "get_workflow_logs", workflow=workflow_name
-                )
-            )["logs"]
+        async with aiof.open(savedir / filename, "wb") as f:
+            await f.write(data)
 
-            if workflow_state.success is False:
-                logs = f"{workflow_state.error_message}{logs}"
-            experiment_dirpath.joinpath(LOGS_FILENAME).write_text(
-                logs, encoding="utf-8"
+        return savedir / filename
+
+    async def list_files(self, experiment_run: ExperimentRun) -> list[FileDetail]:
+        files = await self._generic_reana_call(
+            "list_files", workflow=experiment_run.workflow_name
+        )
+        return [
+            FileDetail(
+                filepath=file["name"],
+                size=file["size"]["raw"],
+                last_modified=file["last-modified"],
             )
-        except ReanaConnectionException as e:
-            raise e
-        except Exception:
-            # TODO: Handle exception properly
-            pass
+            for file in files
+        ]
 
-        metrics_filepath = f"{RUN_OUTPUT_FOLDER}/{METRICS_FILENAME}"
-        await self.download_files(
-            experiment_run,
-            workflow_filepath=metrics_filepath,
-            local_save_dirpath=experiment_dirpath,
+    async def _generic_reana_call(self, function_name, *args, **kwargs):
+        return await asyncio.to_thread(
+            self.__reana_call, function_name, *args, **kwargs
         )
-        return True
 
-    async def _call_reana_function(self, function_name, *args, **kwargs):
-        if not await self.ping():
+    def _ping(self) -> bool:
+        for _ in range(5):
+            out = client.ping(access_token=settings.REANA_ACCESS_TOKEN)
+            if out["error"] is False:
+                return True
+        return False
+
+    def __reana_call(self, function_name, *args, **kwargs):
```

Comment:
I think the name of this function should start with a single underscore. Additionally, it is not clear what is the difference between `_reana_call` and `_generic_reana_call`. Perhaps we can rename them to something like `_(generic_)reana_call` and `_(generic_)async_reana_call`.